Update fairwinds-metrics

mjhuber opened the pull request:
Update fairwinds metrics to v0.4.0 and add support for event listener that was added in this release.

sudermanjr commented on incubator/fairwinds-metrics/templates/controller-deployment.yaml:
```diff
@@ -53,6 +53,14 @@ spec:
+            - --event-listener=true
+            - --events-port={{ .Values.listener.port }}
+            - --events-username={{ .Values.listener.username }}
+            - --events-password={{ .Values.listener.password | quote }}
```
Would it be possible to pass this as a secretRef and pull from env instead? I'm not a fan of passing secrets to helm in plain text.

mjhuber replied:
sure, will take a few updates to the app

mjhuber replied:
i added a secret & moved the username/password there.  This requires https://github.com/FairwindsOps/custom-metrics/pull/19 to be merged.  I also updated the image tag to prep.

sudermanjr commented on incubator/fairwinds-metrics/values.yaml:
```diff
@@ -36,3 +36,27 @@ controller:
     port: 10042
+
+listener:
+  enabled: false
```
Can you please add a ci/test-values that enables this listener so that we can e2e test it?

mjhuber replied:
👍 done

sudermanjr commented on incubator/fairwinds-metrics/README.md:
```diff
@@ -26,3 +26,14 @@ A controller for a collection of non-standard metrics.
+| `listener.ingress.hosts` | listener ingress hosts | | |
+| `listener.ingress.tls` | listener ingress tls config | | |
+| `listener.service.type` | | `ClusterIP` | | |
+| `listener.service.port` | | | | |
```
Can you please add the default for this value?

sudermanjr commented on incubator/fairwinds-metrics/templates/controller-deployment.yaml:
```diff
@@ -65,6 +73,11 @@ spec:
               protocol: TCP
+          {{- if .Values.listener.enabled }}
+            - name: listener
+              containerPort: {{ .Values.listener.port }}
```
Probably overkill to make this configurable, but I'm almost never against more configurability.

mjhuber commented:
Updated to address comments above:
* added ci test-values
* put the listener username/password in a secret
* updated readme documentation

the changes to using env variables is dependent on this PR getting merged and released: https://github.com/FairwindsOps/custom-metrics/pull/19

sudermanjr commented:
I think that PR is ready to go. Do you want to release a new custom-metrics and incorporate that into this PR?

mjhuber commented:
> I think that PR is ready to go. Do you want to release a new custom-metrics and incorporate that into this PR?

sure, a release has been created & I added the option to the chart to use an existing secret for credentials.
